### PR TITLE
[skills] update genai skills

### PR DESCRIPTION
**Pull request by Mohamed-Ashraf273**
## Description

Update genai skills

## Checklist:
- [x] This PR follows [GenAI Contributing guidelines](https://github.com/openvinotoolkit/openvino.genai?tab=contributing-ov-file#contributing). 
- [ ] Tests have been updated or added to cover the new code. 
- [x] This PR fully addresses the ticket. 
- [ ] I have made corresponding changes to the documentation.

### CODE REVIEW

File: .github/agents/model-enabler.index.json [hunk 1]
```diff
@@ -0,0 +1,60 @@
+{
```
Comment (as-suvorov) [inline .github/agents/model-enabler.index.json]: Let's remove index.json files. And corresponding comments from the skills.

Comment (Mohamed-Ashraf273) [reply .github/agents/model-enabler.index.json]: Done!

**Review by rkazants — COMMENTED** [review]
@sbalandi, @as-suvorov, please review again

**Review by sbalandi — COMMENTED** [review]
wwb LGTM , thank you !
Did you run llm_bench ? Do any clarifying comments are needed to be added there , like for wwb?

**Comment by Mohamed-Ashraf273** [comment]
> wwb LGTM , thank you ! Did you run llm_bench ? Do any clarifying comments are needed to be added there , like for wwb?

Thanks! llm_bench wasn’t run because validation used a local tiny-model path, which bypassed that step. I’ve fixed this so local models also run the full model checker. Do you think similar clarifying instructions should be added to the llm_bench analyzer as well?

File: .github/agents/model-enabler.agent.md [hunk 2]
```diff
@@ -21,7 +21,9 @@ You are the OpenVINO GenAI Architect. Your job is to fully enable a new HuggingF
 
 Expect the user to provide:
 
-- **model_id**: HuggingFace model identifier (e.g. `google/gemma-3-4b-it`)
+- **model_id**: HuggingFace model identifier (e.g. `google/gemma-3-4b-it`),
+  local architecture-preserving tiny-model directory, or exported OpenVINO IR
+  directory
```
Comment (as-suvorov) [inline .github/agents/model-enabler.agent.md]:
```suggestion
- **model_id**: HuggingFace model identifier (e.g. `google/gemma-3-4b-it`), or path to local directory with exported OpenVINO IR model (tiny-random or real weights).
```

File: .github/agents/model-enabler.agent.md [hunk 3]
```diff
@@ -44,6 +46,18 @@ Ensure the Python virtual environment is activated before running any commands.
...
+checker path that requires a Hub ID. Instead:
+
+1. Read its `config.json` and processor metadata to establish `model_type`,
+   architecture, task, and whether it is a source model or exported IR.
+2. Reuse a supplied export when present; otherwise export the local source
+   model into a local work directory.
+3. Run the smallest correct direct GenAI generation check for the task.
+4. Preserve the original Hugging Face identity in the analysis report when it
+   is known, while using the local path as the runnable validation artifact.
+
```
Comment (as-suvorov) [inline .github/agents/model-enabler.agent.md]:
```suggestion
```
This should already be fixed on model_checker script side. Please double check and remove

File: .github/agents/model-enabler.agent.md [hunk 4]
```diff
@@ -72,6 +86,25 @@ Before modifying shared model code, check backward compatibility:
 After enablement, re-run **model-checker** with `--skip-export` to validate the fix.
 If model-checker passes, proceed to Step 4.
 
+Revalidate using the same mode used initially: model-checker for a Hub model,
+or the same direct local-path generation reproducer for a local model/IR. Do
+not replace a failing local-artifact check with success from another model.
```
Comment (as-suvorov) [inline .github/agents/model-enabler.agent.md]: If you want to address several models enablement in parallel let's address it more generally. The better approach would be to use subfolders per model type under `.model_enabler`.

File: .github/agents/model-enabler.agent.md [hunk 5]
```diff
@@ -72,6 +86,25 @@ Before modifying shared model code, check backward compatibility:
...
+
+If any GenAI source file changed, rebuild the edited checkout before running
+model-checker or direct inference. Verify `openvino_genai.__file__` resolves to
+the local build and that its linked OpenVINO libraries match the active
+runtime. A result produced by an installed/prebuilt wheel does not validate
+local source changes.
```
Comment (as-suvorov) [inline .github/agents/model-enabler.agent.md]: vlm-model-enabler skill should ensure current version is installed. Let's address issues there.
```suggestion

```

File: .github/agents/model-enabler.agent.md [hunk 6]
```diff
@@ -72,6 +86,25 @@ Before modifying shared model code, check backward compatibility:
...
+
+For every newly enabled VLM, add the repository test coverage required by the
+`vlm-model-enabler` skill. GenAI source changes without a corresponding
+tiny-random test are incomplete.
```
Comment (as-suvorov) [inline .github/agents/model-enabler.agent.md]: There is ### Step 4: Add Tests section present already
```suggestion

```

File: .github/agents/model-enabler.agent.md [hunk 7]
```diff
@@ -72,6 +86,25 @@ Before modifying shared model code, check backward compatibility:
...
+
+Do not report a model as enabled when its required repository test is blocked,
+failing, deselected, or references an inaccessible tiny-model repository. A
+test merely added to source is not validation. Repair the fixture or report the
+exact blocker and leave enablement incomplete.
```
Comment (as-suvorov) [inline .github/agents/model-enabler.agent.md]: Let's move to section 4

File: .github/agents/model-enabler.agent.md [hunk 8]
```diff
@@ -122,3 +155,7 @@ Report a structured summary:
   - **Details**: summary of doc changes.
   - **PR Created** (if applicable): Link or reference to documentation PR
 - **Details**: Additional details required for context, next steps, or follow-ups.
+
+Do not commit, push, or create a pull request unless the user explicitly asks.
+Leave validated changes in the working tree and report the repository path,
+branch, and changed files.
```
Comment (as-suvorov) [inline .github/agents/model-enabler.agent.md]: There is no instruction to open pr in this agent, so no reason to guard against
```suggestion
```

File: .github/skills/vlm-model-enabler/model-inference-image-text.md [hunk 9]
```diff
@@ -7,6 +7,31 @@ For the list of reusable C++ preprocessing utilities (`clip.hpp` / `clip.cpp`) a
 
 ## Test Script Templates
 
+### Smoke Test — `test_image_text.py`
```
Comment (as-suvorov) [inline .github/skills/vlm-model-enabler/model-inference-image-text.md]: We already have `test_image_text_compare.py`. Please explain why dedicated test needed

File: .github/skills/vlm-model-enabler/model-inference-text-to-text.md [hunk 10]
```diff
@@ -3,6 +3,22 @@
 This file provides **supplementary details** for Step 2 of the VLM Model Enabler skill.
 The main workflow is in [SKILL.md](SKILL.md). Do not follow this file as a standalone procedure.
 
+### Smoke Test — `test_text_only.py`
```
Comment (as-suvorov) [inline .github/skills/vlm-model-enabler/model-inference-text-to-text.md]: Same here, please explain

File: .github/skills/vlm-model-enabler/SKILL.md [hunk 11]
```diff
@@ -82,10 +82,22 @@ Follow the checklist in [genai-vlm-architecture.md](genai-vlm-architecture.md) "
 
 ### 2.2 Build
 
+Build the current checkout using the repository instructions in
```
Comment (as-suvorov) [inline .github/skills/vlm-model-enabler/SKILL.md]: Most of the time no special openvino version is needed for local development. Make `pip install .` default option. Instruct how to build with ov sources when required only

File: .github/skills/vlm-model-enabler/SKILL.md [hunk 12]
```diff
@@ -117,22 +129,95 @@ Replace the VisionEncoder stub. Key references:
 
 Update `get_inputs_embeds()` to handle the non-empty images case: insert vision embeddings at placeholder token positions.
 
+Before reporting a tokenizer/export blocker, compare Hugging Face and OpenVINO
+tokenization for every required image/chat special token. If the OpenVINO
+tokenizer drops or splits added tokens, implement architecture-aware recovery
```
Comment (as-suvorov) [inline .github/skills/vlm-model-enabler/SKILL.md]: What is `architecture-aware recovery` ? Why it's being addressed in GenAI agent?

File: .github/skills/vlm-model-enabler/SKILL.md [hunk 13]
```diff
@@ -117,22 +129,95 @@ Replace the VisionEncoder stub. Key references:
...
+export merely because the command requested fp32.
+
 ### Checkpoint
 
 - [ ] Build succeeds
 - [ ] `.model_enabler/test_image_text_compare.py` shows semantically similar outputs
 
 ---
 
+## Step 4 — Repository Test Coverage
```
Comment (as-suvorov) [inline .github/skills/vlm-model-enabler/SKILL.md:193]: Let's move test section from agent to this skill.

**Review by as-suvorov — CHANGES_REQUESTED** [review]

File: .github/agents/model-enabler.agent.md (hunk 3 above)
Comment (Mohamed-Ashraf273) [reply .github/agents/model-enabler.agent.md]: Removed the duplicated local-path handling and updated revalidation to use model-checker for both Hub IDs and local OpenVINO IR directories. Thanks!

File: .github/agents/model-enabler.agent.md (hunk 5 above)
Comment (Mohamed-Ashraf273) [reply .github/agents/model-enabler.agent.md]: Removed the duplicated instructions from the main agent. The vlm-model-enabler skill already requires building the current checkout, verifying openvino_genai.__file__, and checking runtime compatibility. Thanks!

File: .github/agents/model-enabler.agent.md (hunk 4 above)
Comment (Mohamed-Ashraf273) [reply .github/agents/model-enabler.agent.md]: Good point. I updated the model-checker and VLM enablement skills to keep each model type’s artifacts under `.model_enabler/<model_type>/`, so parallel and resumed runs remain isolated. Thanks!

File: .github/agents/model-enabler.agent.md (hunk 6 above)
Comment (Mohamed-Ashraf273) [reply .github/agents/model-enabler.agent.md]: Removed the duplicated test-coverage instruction and kept the detailed requirements in Step 4. Thanks!

File: .github/agents/model-enabler.agent.md (hunk 7 above)
Comment (Mohamed-Ashraf273) [reply .github/agents/model-enabler.agent.md]: Moved this test-validation requirement into Step 4 so all test-related guidance is kept together. Thanks!

File: .github/agents/model-enabler.agent.md (hunk 8 above)
Comment (Mohamed-Ashraf273) [reply .github/agents/model-enabler.agent.md]: Removed the unrelated commit/PR guard since this agent does not include PR creation. Thanks!

File: .github/skills/vlm-model-enabler/model-inference-image-text.md (hunk 9 above)
Comment (Mohamed-Ashraf273) [reply .github/skills/vlm-model-enabler/model-inference-image-text.md]: You’re right. The comparison script already runs the GenAI image-generation path and adds validation against Optimum, so the dedicated smoke test was redundant. I removed it. Thanks!

File: .github/skills/vlm-model-enabler/model-inference-text-to-text.md (hunk 10 above)
Comment (Mohamed-Ashraf273) [reply .github/skills/vlm-model-enabler/model-inference-text-to-text.md]: Same rationale here.
The comparison test already covers the smoke-test path, so I removed the duplicate. Thanks!

File: .github/skills/vlm-model-enabler/SKILL.md (hunk 11 above)
Comment (Mohamed-Ashraf273) [reply .github/skills/vlm-model-enabler/SKILL.md]: Updated the skill to use `python -m pip install` . as the default local-development path. OpenVINO source builds are now limited to cases that require them, such as OpenVINO source changes or unresolved ABI/runtime mismatches. Thanks!

File: .github/skills/vlm-model-enabler/SKILL.md (hunk 12 above)
Comment (Mohamed-Ashraf273) [reply .github/skills/vlm-model-enabler/SKILL.md]: By recovery, I mean restoring only the configured added special-token IDs when the OpenVINO tokenizer drops or splits them: tokenize normal text normally, then splice the exact configured IDs back into input_ids. This belongs in the GenAI VLM enablement flow because `get_inputs_embeds()` relies on image-placeholder IDs to locate where vision embeddings are merged. I clarified this wording in the skill.

File: .github/skills/vlm-model-enabler/SKILL.md (hunk 13 above)
Comment (Mohamed-Ashraf273) [reply .github/skills/vlm-model-enabler/SKILL.md:193]: Moved the detailed repository-test workflow into Step 4 of vlm-model-enabler; the main agent now delegates to that section before documentation updates. Thanks!

File: .github/agents/model-enabler.agent.md [hunk 14]
```diff
@@ -70,33 +70,18 @@ Before modifying shared model code, check backward compatibility:
 - Preserve existing behavior. Prefer branching on explicit code-visible capabilities or model contracts instead of broad model-family checks.
 
 After enablement, re-run **model-checker** with `--skip-export` to validate the fix.
-If model-checker passes, proceed to Step 4.
+If model-checker passes, complete Step 4 — **Repository Test Coverage** in the
+**vlm-model-enabler** skill before updating documentation.
```
Comment (as-suvorov) [inline .github/agents/model-enabler.agent.md]: Let's remove this instruction, step 4 already in **vlm-model-enabler** which agent instructed to follow
```suggestion
```

File: .github/skills/vlm-model-enabler/SKILL.md [hunk 15]
```diff
@@ -82,20 +85,40 @@ Follow the checklist in [genai-vlm-architecture.md](genai-vlm-architecture.md) "
 
 ### 2.2 Build
 
+For normal local development, install the current checkout into the active
+environment using the repository instructions in `src/docs/BUILD.md`:
+
+```bash
+python -m pip install .
+```
```
Comment (as-suvorov) [inline .github/skills/vlm-model-enabler/SKILL.md]: You asked to follow build from sources instructions which require openvino sources, then you provide pip install.
```suggestion

\```bash
pip install --pre -U . --extra-index-url https://storage.openvinotoolkit.org/simple/wheels/nightly
\```

When working with OpenVINO modified source files build with OpenVINO sources. Follow build instructions at: `src/docs/BUILD.md`
```

gh editor cannot handle inner code snippets, so remove `\` from "\\```"

File: .github/skills/vlm-model-enabler/SKILL.md [hunk 16]
```diff
@@ -117,16 +140,95 @@ Replace the VisionEncoder stub. Key references:
...
+   `tests/python_tests/test_vlm_pipeline.py`, including the prompt, image or
+   video tag where applicable, resolution, and any targeted skip/xfail entry
+   required by an already tracked issue.
+3. Add a dedicated `tests/python_tests/test_<model_type>_*.py` only when the
+   behavior cannot fit the shared VLM suite. Reuse existing fixtures and
+   conversion-cache helpers; never use a full-size model in repository tests.
+4. Run the narrow pytest selection for the new model from the activated
+   environment.
+5. If special Transformers or Optimum dependencies are required, add a
+   dedicated VLM CI matrix entry instead of changing unrelated jobs.
```
Comment (as-suvorov) [inline .github/skills/vlm-model-enabler/SKILL.md]:
```suggestion
1. **Find the tiny-random model id**:
   - First infer it from the optimum-intel model description, tests, or release notes when available.
   - If it is not documented there, look it up directly on HuggingFace Hub.
   - Matching the `optimum-intel-internal-testing/tiny-random-*` prefix, for example `optimum-intel-internal-testing/tiny-random-gemma4-unified-it`.
2. **Add the model to VLM Python tests**:
   - Prefer extending `tests/python_tests/test_vlm_pipeline.py` with the tiny-random model id, prompt image tag, video tag if applicable, resolution, and any targeted skip/xfail entry required by an already-tracked issue.
   - Add a dedicated `tests/python_tests/test_<model_type>_*.py` only when the new model requires behavior that does not fit the shared VLM pipeline suite.
   - Use the existing VLM fixtures and converted-model cache helpers; do not use the full-size model in repository tests.
3. **Validate locally**:
   - Run the narrow pytest target for the added or modified tests from the activated virtual environment.
   - If the test cannot run locally, document the exact command, blocker, and expected CI coverage.
```

**Review by as-suvorov — CHANGES_REQUESTED** [review]

File: .github/agents/model-enabler.agent.md (hunk 14 above)
Comment (Mohamed-Ashraf273) [reply .github/agents/model-enabler.agent.md]: Thanks, removed the duplicated instruction from the agent. Repository test coverage remains in Step 4 of the `vlm-model-enabler` skill, which the agent is already instructed to follow.

File: .github/skills/vlm-model-enabler/SKILL.md (hunk 15 above)
Comment (Mohamed-Ashraf273) [reply .github/skills/vlm-model-enabler/SKILL.md]: Thanks, you’re right.
The previous wording mixed the normal wheel-based setup with the OpenVINO source-build path. I’ve separated the two cases and fixed the code fence formatting.

**Review by as-suvorov — APPROVED** [review]

**Comment by as-suvorov** [comment]
build_jenkins

**Review by sbalandi — APPROVED** [review]

**Comment by as-suvorov** [comment]
build_jenkins
